File: microservice/pkg/common/models/task.go
package models

import (
	"gorm.io/gorm"
)

type Task struct {
    gorm.Model  // adds ID, created_at etc.
    Name        string `json:"name"`
    Description string `json:"description"`
    DueDate     string `json:"due_date"`
    Status      string `json:"status"`
}

File: server/src/models/Task.ts
export interface ITask {
    name: string;
    description: string;
    dueDate: string;
}

export interface ITaskModel extends ITask{

}

export class Task implements ITask{

    name: string;
    description: string;
    dueDate: string;

    constructor(name: string, description: string, dueDate: string) {
        this.name = name;
        this.description = description;
        this.dueDate = dueDate;
    }
}
